sql/core/src/test/resources/sql-tests/analyzer-results/ansi/interval.sql.out

```diff
@@ -983,7 +983,7 @@ org.apache.spark.sql.catalyst.parser.ParseException
 {
   "errorClass" : "_LEGACY_ERROR_TEMP_0063",
   "messageParameters" : {
-    "msg" : "Interval string does not match day-time format of `[+|-]d h`, `INTERVAL [+|-]'[+|-]d h' DAY TO HOUR` when cast to interval day to hour: 20 15:40:32.99899999, set spark.sql.legacy.fromDayTimeString.enabled to true to restore the behavior before Spark 3.0."
+    "msg" : "[INTERVAL_ERROR.UNMATCHED_FORMAT_STRING] Interval error. Interval string does not match day-time format of `[+|-]d h`, `INTERVAL [+|-]'[+|-]d h' DAY TO HOUR` when cast to interval day to hour: 20 15:40:32.99899999, set spark.sql.legacy.fromDayTimeString.enabled to true to restore the behavior before Spark 3.0.. SQLSTATE: 22009"
```

Comment:
Can you try to avoid/fix the situation when an exception with properly assigned name for error condition is included to a legacy one?

Reply:
These are the tests that are using golden files. They are used in `SQLQueryTestSuite` and there are quite a lot of them which are using the same logic. I'm not sure if I can fix all of them(this is not the only one that is checked like this)

Reply:
Let's bypass SparkThrowable here:
https://github.com/apache/spark/blob/30152d0ee938d64cadf8d4acd72115750f424e3b/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala#L3402-L3408
like:
```scala
        case st: SparkThrowable => throw st
        // Handle Exceptions thrown by CalendarInterval
        case e: IllegalArgumentException =>
```